apiservers/engine/API-COMPATIBILITY.md

```diff
@@ -10,44 +76,694 @@ GET /containers/json
+* 506 : impossible to attach
+* 500 : server error
+
+## Inspect a container
+
+``` GET /containers/(id)/json ```
+
+This remote API returns low level metadata for the container. This includes almost all of what is previded at the time a create call
+is invoked. This is a __supported API__.
+
+### Status Codes
+
+* 200 : no error
+* 404 : no such container
+* 500 : no error
+
+## List processes running inside a container
+
+```GET containers/(id)/top```
+
+__Not Supported By VIC__
+
+## Get container logs
+
+```GET /containers/(id)/logs```
+
+This operation will return StdOut and StdErr logs from the target container. __Supported in VIC__
+
+### Query parameters
+
+|Paramter|Description|support|
+|---|---|---|
+| __follow__ |1/true/True or 0/false/False, defaults to false. This determines if a stream is returned. | YES |
+| __stdout__ |1/true/True or 0/false/False, defaults to false. Indicates whether stdout logs are desired. | YES | 
+| __stderr__ | 1/true/True or 0/false/False, defaults to false. Indicates whether stderr logs are desired.| YES | 
+| __since__ | integer based UNIX timestamp with which to filter logs. Only logs since the timestamp will be returned. Default: 0. | *maybe* |
+| __timestamps__ | 1/true/True or 0/false/False, defaults to false. timestamped log lines. Defaults to false.| *maybe* |
+| __tail__ | all or <number>. indicates how many lines to output since the end of the logs. | *maybe* |
+
+### Status Codes 
+
+* 101 : no error, indicates proxy about hijacking
+* 200 : no error, no upgrade header from attach found
+* 404 : no such container
+* 500 : server error
+
+## Inspect changes on a container’s filesystem
+
+```GET /containers/(id)/changes```
+
+inspects the r/w layer of a container for changes to it's filesystem.
+
+
+
+### Response
+
+the response for this api involves a json based HTTP response that is a json array of objects.
+
+__JSON OBJECT STRUCTURE__
+
+```
+{
+    "Path" : <string, path to file>,
+    "Kind" : <integer, kind code>
+}
+```
+
+__Kind Codes__
+
+|code number| description|
+|---|---|
+|0|Modify|
+|1|Add|
+|2|Delete|
+
+### Response Status Codes
+
+* 200 : no error
+* 404 : no such container
+* 500 : server error
+
+## Export a container
+
+```GET /containers/(id)/export```
+
+This exports the contents of a container as a flattened file for other than docker use. 
+__VIC plans to support this in the future__
+
+### Response
+
+HTTP response that returns a binary stream of the flattened file in a tarball. The response is of type application/octet-stream. 
+
+### Response Status Codes
+
+* 200 : no error
+* 404 : no such container
+* 500 : server error
+
+## Get container stats based on resource usage
+```GET /containers/(id)/stats```
+
+This api returns a stream of the containers resource usage.
+
+__Not Supported BY VIC__
+
+the isolation unit of a containerVM is a vm so these stats are actively available from vSphere.
+
+## Resize a container TTY
+
+__Not Supported by VIC__
```

Comment:
Based on the hack up by George today, we're going to have this support in the tether.  We should probably add this support as it's considered a measure of quality for attach.